src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -4441,8 +4441,13 @@ checkNullable:
                     ' being reported. For now keep backwards compatibility.
                     If param.ContainsDiagnostics Then
                         param = param.AddTrailingSyntax(ResyncAt({SyntaxKind.CommaToken, SyntaxKind.CloseParenToken}))
-                    End If
+                    Else
+                        ' TODO: MAKE THIS CONDITIONAL ON A LANGUAGE FEATURE FLAG
+                        If param.Default Is Nothing Then
+                            param = param.AddTrailingSyntax(ResyncAt({SyntaxKind.CommaToken, SyntaxKind.CloseParenToken}))
```

Comment:
Simply re-syncing without producing any diagnostics about unexpected things doesn't feel like a right thing to do.  Do we have a test case for a scenario like ```(Optional x as Integer 1)```?